tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/eval/ParseBank.scala

```diff
@@ -0,0 +1,65 @@
+package org.allenai.nlpstack.parse.poly.eval
+
+import org.allenai.nlpstack.parse.poly.polyparser._
+
+/** A ParseBank maps string-representations of sentences to a unique parse.
+  *
+  * @param underlyingMap the map from strings to parses
+  */
+case class ParseBank(underlyingMap: Map[String, PolytreeParse]) {
+
+  /** Takes a candidate parse tree and finds the parse in the bank corresponding to its
+    * sentence. If no parse exists (or if the existing parse has a different tokenization than
+    * the candidate parse), then None is returned.
+    *
+    * @param candParse the parse, for which we want to find a corresponding banked parse
+    * @return the corresponding banked parse, if it exists
+    */
+  def askForGoldParse(candParse: PolytreeParse): Option[PolytreeParse] = {
```

Comment:
I find looking up a gold parse by a candidate parse somewhat odd. There is a gold parse for each sentence. So why not look up just based on sentence and move this filtering logic to check mismatched tokens as a helper into `ParseAnalyzer`?